src/controller/ReadInteraction.h

```diff
@@ -45,7 +45,10 @@ template <typename DecodableAttributeType>
 CHIP_ERROR ReadAttribute(Messaging::ExchangeManager * aExchangeMgr, const SessionHandle sessionHandle, EndpointId endpointId,
                          ClusterId clusterId, AttributeId attributeId,
                          typename TypedReadAttributeCallback<DecodableAttributeType>::OnSuccessCallbackType onSuccessCb,
-                         typename TypedReadAttributeCallback<DecodableAttributeType>::OnErrorCallbackType onErrorCb)
+                         typename TypedReadAttributeCallback<DecodableAttributeType>::OnErrorCallbackType onErrorCb,
```

Comment:
could we set some of these arguments as some 'params' object?

This  method now has 11 arguments out of which 2 are just 'integers' and one is a bool (so it is generally hard to read `Foo(true)` and figure out what true actually does, like `ReadAttribute(...., true, ...)` nobody will obviously figure out that true means keepSubscription.

Reply:
This method is not really the public API (the templates over the attribute info are; this is a utility helper), but more generally it doesn't make sense to put subscriptions into an API called ReadAttribute.

We should have a Subscribe public API that perhaps calls this method (which could get renamed) and takes subscription info as a params struct or something.